### ISSUE TITLE
FFmpeg and libopus

### ISSUE THREAD

**Issue by bnnm** [kode54/vgmstream#321]
Paging @kode54 

FFmpeg's internal opus decoder is audibly glitchy with some low bitrate files. The same files forced through FFmpeg's libopus (ex. `ffmpeg.exe -codec libopus -i in.opus out.wav`), or regular (foobar's) libopus sound fine (plus libopus is faster). I tried a bunch of fixes but I think it's just FFmpeg's opus that is buggy.

I see the DLL builder you use ([here](https://github.com/jb-alvarado/media-autobuild_suite)) has a libopus option, so I hope you could make a build using it. The appropriate options should be: add `--enable-libopus`, change `--enable-decoder=opus` to `--enable-decoder=libopus`.

If it's a lot of trouble I could try other ways like integrating libopus.dll directly, but seems that'd be a lot of effort so it'd be easier this way.

**Comment by richardpl** [comment]
@bnnm do you have opus files which are glitchy?

**Comment by bnnm** [comment]
@richardpl - example: https://www.sendspace.com/file/e3z8e0

@kode54 - also, if you could add: `--enable-decoder=mp1float,mp2float`, and I think you can remove `--enable-decoder=bink` (not binkaudio or demuxer tho)

**Comment by shockdude** [comment]
I can confirm that rebuilding the dlls with --enable-libopus and --enable-decoder=libopus decreases the glitchiness of vgmstream's opus playback.

**Comment by richardpl** [comment]
@bnnm encoder delay? do you mean skip samples at beginning? I fixed that bug recently.

**Comment by bnnm** [comment]
@kode54 - thanks a lot, it's perfect now.

@richardpl - yes, "encoder delay" is a name used elsewhere for those skip samples. Great it's finally patched, but still, it took ~4 years since the decoder was added, even though it's easy to test vs opusdec.exe.

Several other MDCT codecs also are missing skip samples (or padding/end samples) since forever, it's all very unfortunate FFmpeg devs have little interest in proper audio decoding. I've posted trac issues and sent patches that were mostly ignored (remember I had to pretty much beg you to apply my last XMA patch).

vgmstream patches delay on the fly though, so it's no big deal for me. Just take it as a friendly suggestion to raise to other devs.
